tests/unit/App.spec.ts

```diff
@@ -1,37 +1,99 @@
 import myAuthenticationPlugin from 'authenticationPlugin/App'
+import firebase from 'firebase/app'
+
 jest.mock('firebase/app', () => {
-    const firebase = {
-        auth: jest.fn(() => {
-            return {
-                currentUser: {
-                    email: 'test',
-                    uid: '123',
-                    emailVerified: <boolean>true
-                },
-            }
-        }),
-        initializeApp: jest.fn()
-    }
-    return firebase
-})
+  const userCredentialMock = {
+    user: {
+      sendEmailVerification: jest.fn(),
+    },
+  };
+  return {
+    auth: jest.fn().mockReturnThis(),
+    currentUser: {
+      email: 'test',
+      uid: '123',
+      emailVerified: true
+    },
+    signInWithEmailAndPassword: jest.fn(),
+    createUserWithEmailAndPassword:jest.fn(() => userCredentialMock),
+    sendPasswordResetEmail:jest.fn(),
+    signOut:jest.fn(),
+    onAuthStateChanged:jest.fn(),
+    initializeApp:jest.fn()
+  };
+});
 describe('App.js (Authentication Plugin)', () => {
-    describe('Tests for getLoggedInUser ()', () => {
-        it('The API should return the email of the currently logged in user', () => {
-            expect(myAuthenticationPlugin.getLoggedInUser().email).toBe('test')
-        })
+  const email = 'test'
+  const password = 'mypassword'
+  afterAll(() => {
+    jest.resetAllMocks();
+  });
+  describe('Tests for getLoggedInUser ()', () => {
+    it('The API should return the email of the currently logged in user', () => {
+      expect(myAuthenticationPlugin.getLoggedInUser().email).toBe('test')
+    })
+
+    it('The API should return the unique userID for the currently logged in user', () => {
+      expect(myAuthenticationPlugin.getLoggedInUser().userId).toBe('123')
+    })
 
-        it('The API should return the unique userID for the currently logged in user', () => {
-            expect(myAuthenticationPlugin.getLoggedInUser().userId).toBe('123')
-        })
+    it('The API should return a boolean value telling us, If the email is verified or not ', () => {
+      expect(myAuthenticationPlugin.getLoggedInUser().isEmailVerified).toBe(true)
+    })
```

Comment:
there are all the same test case. i dont' understand why they have to be different tests